### PR TITLE
chore: use a single node for all insights in notebooks

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We currently have two separate nodes that call `Query` under the hood

## Changes

Make it so that both types of insights become the same node:
- Move the paste rule from `NotebookNodeInsight` to `NotebookNodeQuery`
- Set the title of the Query node to that of the saved insight
- Delete the unnecessary `NotebookNodeInsight` node

## How did you test this code?

By eye

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
:shipit:

**Comment by pauldambra** [comment]
Ah, wait... any existing notebook with an insight in it is invalid if we merge this...

Screenshot 2023-09-03 at 21 16 16

Should we care? Could we DB migrate them or migrate them on load?

File: frontend/src/scenes/data-warehouse/viewLinkLogic.tsx [hunk 1]
```diff
@@ -8,7 +8,6 @@ import { databaseSceneLogic } from 'scenes/data-management/database/databaseScen
 import { loaders } from 'kea-loaders'
```
Comment (EDsCODE) [inline frontend/src/scenes/data-warehouse/viewLinkLogic.tsx:8]: Was there a blocking reason these changes were reverted?

REF: https://github.com/PostHog/posthog/pull/17287

Comment (daibhin) [reply frontend/src/scenes/data-warehouse/viewLinkLogic.tsx:8]: Ok yikes! None whatsoever. I think that was just a terrible merge on my part. Sorry about that Eric. I see it's back in now

**Pull request merged**